src/ParametricOptInterface.jl

```diff
@@ -691,18 +711,38 @@ end
 function add_constraint_with_parameters_on_function(
     model::Optimizer,
     f::MOI.ScalarAffineFunction{T},
-    set::MOI.AbstractScalarSet,
-) where {T}
+    set::S,
+) where {T,S}
     vars, params, param_constant =
         separate_possible_terms_and_calculate_parameter_constant(model, f.terms)
-    ci = MOI.Utilities.normalize_and_add_constraint(
-        model.optimizer,
-        MOI.ScalarAffineFunction(vars, f.constant + param_constant),
-        set,
-    )
-    model.affine_constraint_cache[ci] = params
-    model.original_constraint_function_and_set_cache[ci] = (f, set)
-    return ci
+    model.last_affine_added += 1
+    if model.interpret_as_bound_if_possible &&
```

Comment:
I dont think we need the _is_possible piece.
I think that is error prone, in the sense one might be trying to add bounds and silently adding regular constrants.
Then if in as_bound state and the data does not qualify as bound then add throw error.